Feature basic plugin

tntmeijs opened the pull request:
I have removed the default plugin code and replaced it with a very simplistic viewport renderer override.
The plug-in now shows up in under the "Renderer" tab in Maya

The "initializePlugin" and "uninitializePlugin" functions do not use the Wisp coding guideline names, because Maya expects a certain function name. Using capitals will result in Maya failing to initialize the plug-in. Trying to use capitals will result in the error: `initializePlugin function could not be found in plug-in (MayaRayTracedViewport)`.

**Results:**
![maya_plugin_loaded](https://user-images.githubusercontent.com/15856620/48907262-38fdcf00-ee67-11e8-81d1-b7c8f7f20721.png)

tntmeijs committed:
Basic viewport renderer override

[ADD] Code to override the viewport renderer.
[ADD] Plugin initialization.

tntmeijs committed:
Better plug-in name

[CHANGE] Name of the plug-in.

tntmeijs committed:
Fixed initialization failure

[FIX] Null pointer to the rendering instance, the creation code was accidentally removed during refactoring.

VZout commented on src/Defines.hpp:
```diff
@@ -0,0 +1,9 @@
+#define DEFINES_HPP
+
+// Settings for this plug-in
+#define COMPANY_NAME "Team Wisp"
```
prefer static const constexpr

VZout commented on src/main.cpp:
```diff
@@ -1,10 +1,77 @@
+
+// NOTE:	This function name deviates from the coding standard because Autodesk Maya looks for a function that is
+//			written exactly like this.
+MStatus initializePlugin(MObject t_object)
```
Our style guide uses PascalCase not camelCase for functions

tntmeijs replied:
I am aware of it, but Maya requires the function to be called "initializePlugin" and "uninitializePlugin", the way I understand it.

![screenshot](https://user-images.githubusercontent.com/15856620/48908516-c55dc100-ee6a-11e8-8960-a37634521c0f.png)

VZout commented on src/plugin/ViewportRendererOverride.hpp:
```diff
@@ -1,16 +1,24 @@
 #ifndef VIEWPORT_RENDERER_OVERRIDE_HPP
```
Prefer pragma once

tntmeijs committed:
Fixing defines

[CHANGE] Defines into const constexpr.
[CHANGE] "Defines.hpp" into "Settings.hpp".

tntmeijs committed:
Constants wrapped in a class

[CHANGE] "Settings.hpp" into "Constants.hpp".
[CHANGE] Constants in the file are now wrapped in a class.
[CHANGE] Main has been fixed to work with the new constants class.

tntmeijs committed:
Pragma once

[REMOVE] Include guards and changed it with pragma once.

VZout commented on src/Constants.hpp:
```diff
@@ -0,0 +1,13 @@
+
+namespace wisp
+{
+	class Constants
```
This doesn't need to be in a class. a namespace should be used since it doesn't create a default constructor.

tntmeijs replied:
Yup, I accidentally put that commit on the other branch. This will be fixed in the next pull request.

VZout reviewed (commented):
Looks good. Don't see anything wrong with it.
